### PR TITLE
[R] Initial implementation

### PR DESCRIPTION
**Pull request by cyrusmsk**
Simple R implementation.
Using 'fastmap' for hashmap, because R doesn't have proper built-in dictionary data structure

### CODE REVIEW

**Comment by jinyus** [comment]
Getting this error in docker:
```
Rscript related.R 
Error in contrib.url(repos, type) : 
  trying to use CRAN without setting a mirror
Calls: install.packages -> startsWith -> contrib.url
Execution halted
```

**Comment by jinyus** [comment]
I tried to do this manually...it tries to download https://cran.r-project.org/CRAN_mirrors.csv but the url is down.:
```
> echo "chooseCRANmirror()" > setup.R
> Rscript setup.R

Warning: failed to download mirrors file (cannot open URL 'https://cran.r-project.org/CRAN_mirrors.csv'); using local file '/usr/share/doc/R//CRAN_mirrors.csv'
Error in file(file, "rt") : cannot open the connection
Calls: chooseCRANmirror ... getCRANmirrors -> .getMirrors -> read.csv -> read.table -> file
In addition: Warning messages:
1: In download.file(url, destfile = f, quiet = TRUE) :
  URL 'https://cran.r-project.org/CRAN_mirrors.csv': Timeout of 60 seconds was reached
2: In file(file, "rt") :
  cannot open file '/usr/share/doc/R//CRAN_mirrors.csv': No such file or directory
Execution halted

```

**Comment by jinyus** [comment]
> > I tried to do this manually...it tries to download https://cran.r-project.org/CRAN_mirrors.csv but the url is down.
> 
> I've added extra info for installation. But probably the problem that CRAN is down https://www.isitdownrightnow.com/cran.r-project.org.html

Yea bad timing: https://mirrors.ustc.edu.cn/CRAN/

> Dear users,

> The CRAN Admin Team will perform system upgrades during the period Wednesday November 15 until Thursday November 16, 2023.
> There will be intermittent outages in service during this time.
> We thank you for your understanding and apologise for any inconvenience this may cause.

> CRAN Admin Team

**Comment by jinyus** [comment]
Is there anyway to remove some of the unnecessary deps? It's taking long to compile

```
also installing the dependencies ‘utf8’, ‘sys’, ‘prettyunits’, ‘pillar’, ‘pkgconfig’, ‘rappdirs’, ‘proxy’, ‘askpass’, ‘rprojroot’, ‘crayon’, ‘fs’, ‘pkgbuild’, ‘diffobj’, ‘fansi’, ‘rematch2’, ‘tibble’, ‘base64enc’, ‘cachem’, ‘memoise’, ‘sass’, ‘fastmap’, ‘stringi’, ‘e1071’, ‘wk’, ‘curl’, ‘mime’, ‘openssl’, ‘R6’, ‘cli’, ‘glue’, ‘lifecycle’, ‘rlang’, ‘brio’, ‘callr’, ‘desc’, ‘digest’, ‘ellipsis’, ‘evaluate’, ‘magrittr’, ‘pkgload’, ‘praise’, ‘processx’, ‘ps’, ‘waldo’, ‘withr’, ‘highr’, ‘xfun’, ‘yaml’, ‘bslib’, ‘fontawesome’, ‘htmltools’, ‘jquerylib’, ‘stringr’, ‘tinytex’, ‘R.methodsS3’, ‘R.oo’, ‘R.utils’, ‘R.cache’, ‘classInt’, ‘DBI’, ‘Rcpp’, ‘s2’, ‘units’, ‘httr’, ‘vctrs’, ‘testthat’, ‘knitr’, ‘rmarkdown’, ‘R.rsp’, ‘sf’
```

**Comment by cyrusmsk** [comment]
It is possible to User other CRAN mirror.
About dependencies - I'm not sure.. hard to say which dependencies are required.
There are packages in AUR:
https://aur.archlinux.org/packages/r-fastmap
https://aur.archlinux.org/packages/r-jsonlite
So we need to Install them only once..
Maybe move package installation into Docker side? Instead of R script?

**Comment by jinyus** [comment]
> Maybe move package installation into Docker side? Instead of R script?

Sounds good. Because it has been installing for more than 10mins now

File: r/related.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  }
+  
+  #top_indices <- order(-tagged_post_count)[1:TopN]
+  
+  top_related <- jsonText[top_indices,]
+  
+  return(list("_id" = jsonText$`_id`[[idx]], "tags" = jsonText$tags[[idx]], "related" = top_related))
+}
+
+jsonText <- fromJSON("posts.json")
```
Comment (jinyus) [inline r/related.R:58]: The input is '../posts.json'

**Comment by cyrusmsk** [comment]
If the installation is going too long - another option is to use renv package. It is similar to Python virtual environments.
So it will create a Lock file and read all dependencies from this file.

**Comment by jinyus** [comment]
I will merge this but exclude from `run_all` for now, until we can get this docker issue sorted.

**Pull request merged**